pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperDataCache.java

```diff
@@ -46,11 +47,32 @@ public ZooKeeperDataCache(final ZooKeeperCache cache) {
         this.cache = cache;
     }
 
-    public T get(final String path) throws Exception {
-        return getWithStat(path).getKey();
+    public CompletableFuture<Optional<T>> getAsync(String path) {
+        CompletableFuture<Optional<T>> future = new CompletableFuture<>();
+        cache.getDataAsync(path, this, this).thenAccept(entry -> {
+            future.complete(entry.map(Entry::getKey));
```

Comment:
Actually, this should be OK, the `zkFuture.complete(null)` is to make sure the `AsyncLoadingCache` will not retain the entry in the map. That `null` value is then converted into an `Optional.empty()` : 

```java
  }).thenAccept(result -> {
            if (result != null) {
                future.complete(Optional.of((Entry<T, Stat>) result));
            } else {
                future.complete(Optional.empty());
            }
```